core/src/main/java/com/linecorp/armeria/common/multipart/BodyPartBuilder.java

```diff
@@ -0,0 +1,102 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common.multipart;
+
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import org.reactivestreams.Publisher;
+
+import com.google.common.collect.Streams;
+
+import com.linecorp.armeria.common.HttpData;
+
+/**
+ * A builder class for creating {@link BodyPart} instances.
+ */
+public final class BodyPartBuilder {
+
+    private static final Multi<HttpData> EMPTY = Multi.empty();
+
+    private BodyPartHeaders headers = BodyPartHeaders.of();
+    private Multi<HttpData> content = EMPTY;
+
+    BodyPartBuilder() {}
+
+    /**
+     * Sets the specified headers for this part.
+     * @param headers headers
+     */
+    public BodyPartBuilder headers(BodyPartHeaders headers) {
+        requireNonNull(headers, "headers");
+        this.headers = headers;
+        return this;
+    }
+
+    // TODO(ikhoon): Add builder methods for content that take `File` and `Path`
+
+    /**
+     * Adds a new body part backed by the specified {@link Publisher}.
+     * @param publisher publisher for the part content
+     */
+    public BodyPartBuilder content(Publisher<HttpData> publisher) {
+        requireNonNull(publisher, "publisher");
+        if (content == EMPTY) {
+            content = Multi.from(publisher);
+        } else {
+            content = Multi.concat(content, publisher);
+        }
+        return this;
+    }
+
+    /**
+     * Adds the specified a new body part content.
+     */
+    public BodyPartBuilder content(String content) {
+        requireNonNull(content, "content");
+        return content(HttpData.ofUtf8(content));
+    }
+
+    /**
+     * Adds the specified new body part contents.
+     */
+    public BodyPartBuilder content(Iterable<? extends CharSequence> contents) {
+        requireNonNull(contents, "contents");
+        final List<HttpData> wrapped = Streams.stream(contents)
+                                              .map(HttpData::ofUtf8)
+                                              .collect(toImmutableList());
+        return content(Multi.from(wrapped));
+    }
+
+    /**
+     * Adds the specified {@link HttpData} as a part content.
+     */
+    public BodyPartBuilder content(HttpData content) {
+        requireNonNull(content, "content");
+        return content(Multi.singleton(content));
+    }
+
+    /**
+     * Returns a newly-created {@link BodyPart}.
+     */
+    public BodyPart build() {
+        checkState(content != EMPTY, "Should set at lease one content");
```

Comment:
How about requiring the content in the factory method of `BodyPartBuilder`, like other builders we have?

Reply:
That's a good point. `content` is not only a required field but also could be added multiple times.
For example:
```java
BodyPart.builder()
        .headers(...)
        .content("First Line")
        .content("Second Line")
        ...
        .build();
```

I hesitate to move it to the factory method of `BodyPartBuilder`. We can just simplify the content that could be set only one time by removing `BodyPartBuilder`.

```java
BodyPart bodyPartA = BodyPart.of(headers, content);
BodyPart bodyPartB = BodyPart.of(content);
```

Reply:
- Oh, I didn't know a user can specify the content multiple times. Let leave it as it is, then.
- lease -> least

Reply:
> lease -> least

Oops... 😅